packages/application/src/router.ts

```diff
@@ -268,7 +272,7 @@ export class Router implements IRouter {
    * If a pattern is matched, its command will be invoked with arguments that
    * match the `IRouter.ILocation` interface.
    */
-  route(): void {
+  async route(): Promise<void> {
```

Comment:
Does this need to be `async`? You don't seem to have added any `await` calls inside.

```suggestion
  route(): Promise<void> {
```

Reply:
I mean, it is asynchronous, since it's returning a Promise. I suppose that it doesn't need the keyword, but I was trying to be as explicit as possible, since there were a bunch of places in the tests that were `await`-ing the result of this function.

Reply:
I don't think it should matter to consumers of this function whether it was created with `async` or not, at the type level. To them it will look the same, `() => Promise<void>`.

Reply:
Do you think it matters one way or the other?

Reply:
Interesting, I didn't realize the distinction. Do you have a like for that behavior?

Reply:
Okay, I've removed the keyword.

Reply:
I just did some experiments locally, and I seem to be understanding it wrong! The part up to the first `await` in an async function seems to be executed synchronously. For example this code ([in TS playground](https://www.typescriptlang.org/play/index.html#src=async%20function%20is_async()%3A%20Promise<void>%20%7B%0D%0A%20%20%20%20console.log("async")%3B%0D%0A%20%20%20%20await%20Promise.resolve()%0D%0A%20%20%20%20console.log("async%20after%20promise")%3B%0D%0A%7D%0D%0A%0D%0A%0D%0Ais_async()%3B%0D%0Aconsole.log("after%20async")%0D%0A%0D%0A)):

```typescript
async function is_async(): Promise<void> {
    console.log("async");
    await Promise.resolve()
    console.log("async after promise");
}
is_async();
console.log("after async")
```

Has this result:

```
async
after async
async after promise
```

But I can't find any sources talking about whether the first part of the `async` function is executed synchronously.